### PR TITLE
Fixed ci/cd config to test->build->deploy

### LINKED ISSUE #21: CI/CD Pipeline Split Jobs
The pipeline config needs to be structured as follows
- Test phase that runs pytest over repository (On all branches + PRs)
- Build phase that runs linting (On all branches + PRs)
- Deploy phase that builds and deploys duckbot as a Docker container (On merge to main branch)

### PR DESCRIPTION
**Pull request by Chippers255**
- ci/cd phases of test->build->deploy
- bot will now standup and tear down main version of DuckBot on server
- self hosted github actions runner

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -10,30 +10,43 @@ on:
...
-        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        python3.8 -m flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-    - name: Test with pytest
-      run: |
-        pytest -s
+        python3.8 -m flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+  deploy:
+    if: github.event_name == 'push'
```
Comment (twentylemon) [inline .github/workflows/python-package.yml:36]: Can we do `push` event and that push was to the main branch?

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -10,30 +10,43 @@ on:
...
 jobs:
-  build:
-
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: ['3.8']
-
+  test:
+    runs-on: self-hosted
```
Comment (twentylemon) [inline .github/workflows/python-package.yml:14]: Not that it should really matter, but we could run `test` and `build` on github's servers instead (if we have the free tier time). Potentially save some cleanup for your cluster.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM python:3.8
+COPY requirements.txt /
+RUN pip install --upgrade pip
+RUN pip install -r /requirements.txt
```
Comment (twentylemon) [inline Dockerfile]: `python3 -m pip` ?

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (Chippers255) [reply .github/workflows/python-package.yml:36]: Lines 7-10 of the config yaml dictate that the pipeline will only run on main branch. This line makes sure build will only happen on a push to main after merging.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**